packages/paypal-commerce-integration/src/paypal-commerce-accelerated-checkout/paypal-commerce-accelerated-checkout-payment-strategy.ts

```diff
@@ -0,0 +1,351 @@
+    PayPalCommerceConnectStylesOption,
+    PayPalCommerceInitializationData,
+    PayPalCommerceSdk,
+} from '@bigcommerce/checkout-sdk/paypal-commerce-utils';
+
+import PayPalCommerceRequestSender from '../paypal-commerce-request-sender';
+
+import { WithPayPalCommerceAcceleratedCheckoutPaymentInitializeOptions } from './paypal-commerce-accelerated-checkout-payment-initialize-options';
+
+export default class PayPalCommerceAcceleratedCheckoutPaymentStrategy implements PaymentStrategy {
+    private methodId = 'paypalcommerceacceleratedcheckout';
+    private paypalConnectStyles?: PayPalCommerceConnectStylesOption;
+    private paypalConnectCardComponentMethods?: PayPalCommerceConnectCardComponentMethods;
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private paypalCommerceRequestSender: PayPalCommerceRequestSender,
+        private paypalCommerceSdk: PayPalCommerceSdk,
+        private paypalCommerceAcceleratedCheckoutUtils: PayPalCommerceAcceleratedCheckoutUtils,
+    ) {}
+
+    /**
+     *
+     * Default methods
+     *
+     * */
+    async initialize(
+        options: PaymentInitializeOptions &
+            WithPayPalCommerceAcceleratedCheckoutPaymentInitializeOptions,
+    ): Promise<void> {
+        const { methodId, paypalcommerceacceleratedcheckout } = options;
+
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.methodId" argument is not provided.',
+            );
+        }
+
+        if (!paypalcommerceacceleratedcheckout) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.paypalcommerceacceleratedcheckout" argument is not provided.',
+            );
+        }
+
+        if (
+            !paypalcommerceacceleratedcheckout.onInit ||
+            typeof paypalcommerceacceleratedcheckout.onInit !== 'function'
+        ) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.paypalcommerceacceleratedcheckout.onInit" argument is not provided or it is not a function.',
+            );
+        }
+
+        if (
+            !paypalcommerceacceleratedcheckout.onChange ||
+            typeof paypalcommerceacceleratedcheckout.onChange !== 'function'
+        ) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.paypalcommerceacceleratedcheckout.onChange" argument is not provided or it is not a function.',
+            );
+        }
+
+        await this.paymentIntegrationService.loadPaymentMethod(methodId);
+
+        const state = this.paymentIntegrationService.getState();
+        const currencyCode = state.getCartOrThrow().currency.code;
+        const paymentMethod =
+            state.getPaymentMethodOrThrow<PayPalCommerceInitializationData>(methodId);
+
+        this.methodId = methodId;
```

Comment:
Is it necessary to set variable like this? We can avoid it by just passing it as argument for both `runPayPalConnectAuthenticationFlowOrThrow` and `handlePayPalConnectInstrumentChange` functions

Reply:
Yes, you are right. Thanks